source/src/plugins/MeanWithinExpectedTest.cc

```diff
@@ -87,43 +94,106 @@ namespace dqm4hep {
     //-------------------------------------------------------------------------------------------------
 
     StatusCode MeanWithinExpectedTest::readSettings(const TiXmlHandle xmlHandle) {
-      RETURN_RESULT_IF(STATUS_CODE_SUCCESS, !=, XmlHelper::readParameter(xmlHandle, "ExpectedMean", m_expectedMean));
-
-      RETURN_RESULT_IF(STATUS_CODE_SUCCESS, !=,
-                       XmlHelper::readParameter(xmlHandle, "MeanDeviationLower", m_meanDeviationLower,
-                                                [this](const float &value) { return value < this->m_expectedMean; }));
+      RETURN_RESULT_IF(STATUS_CODE_SUCCESS, !=, XmlHelper::readParameters(xmlHandle, "Properties", m_properties));
 
-      RETURN_RESULT_IF(STATUS_CODE_SUCCESS, !=,
-                       XmlHelper::readParameter(xmlHandle, "MeanDeviationUpper", m_meanDeviationUpper,
-                                                [this](const float &value) { return value > this->m_expectedMean; }));
+      RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_FOUND, !=, XmlHelper::readParameter(xmlHandle, "ExpectedMean", m_expectedMean));
+      RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_FOUND, !=, XmlHelper::readParameter(xmlHandle, "Percentage", m_percentage));
+      RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_FOUND, !=, XmlHelper::readParameter(xmlHandle, "MeanDeviationLower", m_meanDeviationLower));
+      RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_FOUND, !=, XmlHelper::readParameter(xmlHandle, "MeanDeviationUpper", m_meanDeviationUpper,
+												       [this](const float &value){return !(isnan(this->m_meanDeviationLower) && isnan(value));}));
 
       return STATUS_CODE_SUCCESS;
     }
 
     //-------------------------------------------------------------------------------------------------
 
     void MeanWithinExpectedTest::userRun(MonitorElementPtr monitorElement, QualityTestReport &report) {
+
       TH1 *pHistogram = monitorElement->objectTo<TH1>();
 
       if (nullptr == pHistogram) {
         report.m_message = "ROOT monitor object is not a TH1 object !";
         throw StatusCodeException(STATUS_CODE_INVALID_PTR);
       }
 
-      const float mean(pHistogram->GetMean());
-      const float range(fabs(m_meanDeviationUpper - m_meanDeviationLower));
-
-      if (m_meanDeviationLower < mean && mean < m_meanDeviationUpper) {
-        report.m_message =
-            "Within expected range: expected " + typeToString(m_expectedMean) + ", got " + typeToString(mean);
-      } 
+      // For-loop code for doing multiple test types per qtest; use TBD
+      /*
+      for (std::vector<std::string>::iterator it = m_properties.begin(); it != m_properties.end(); it++
+      {
+        // Do something
+      }
+      */
+
+      float result;
+
+      if (m_properties[0] == "Mean")
+	{
+	  result = AnalysisHelper::mean(monitorElement, m_percentage);
```

Comment:
You should not use the percentage here, as the user expected a "Mean" test. I thnik having Mean (without percentage), Mean90, RMS, RMS90 and Median is good start.